cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/AbstractProgramWorkflowRunner.java

```diff
@@ -84,13 +86,15 @@ public AbstractProgramWorkflowRunner(Arguments runtimeArguments, RunId runId, Pr
    * @return a {@link Callable} of {@link RuntimeContext} for this {@link Program}
    */
   protected Callable<RuntimeContext> getRuntimeContextCallable(String name, final Program program) {
+    ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+    for (Map.Entry<String, String> systemArgument : systemArguments) {
+      builder.put(systemArgument);
+    }
+    builder.put(ProgramOptionConstants.RUN_ID, runId.getId());
+    builder.put(ProgramOptionConstants.WORKFLOW_BATCH, name);
```

Comment:
So if you look at ``WorkflowDriver``, it received ``ProgramOptions``. From there, it extracted ``logicalStartTime`` via ``programOptions.getArguments().get(...``. Then it passed ``logicalStartTime`` to this class via ``ProgramWorkflowRunnerFactory``. Instead of extracting there, and adding it back here in a new ``BasicArguments`` object, I just pass the original ``BasicArguments`` object around now.